packages/terra-functional-testing/src/services/wdio-selenium-docker-service.js

```diff
@@ -130,85 +70,36 @@ class SeleniumDockerService {
-  async waitForNetworkCreation() {
-    await this.pollCommand('docker network ls | grep wdio', (result) => (
-      new Promise((resolve, reject) => {
-        const { stdout: networkStatus } = result;
-
-        // Resolve if the wdio services are created.
-        if (networkStatus) {
-          resolve();
-        } else {
-          reject();
-        }
-      })));
+  getDockerComposeFilePath() {
+    return path.resolve(__dirname, '../docker/docker-compose.yml');
   }
 
   /**
-   * Ensures the docker services have been created.
+   * Starts the docker selenium hub and waits for the grid to become ready.
    */
-  async waitForServiceCreation() {
-    await this.pollCommand('docker service ls | grep wdio', (result) => (
-      new Promise((resolve, reject) => {
-        const { stdout: serviceStatus } = result;
-
-        // Resolve if the wdio network is created.
-        if (serviceStatus) {
-          resolve();
-        } else {
-          reject();
-        }
-      })));
-  }
-
-  /**
-   * Ensures the docker network has been shut down.
-   */
-  async waitForNetworkRemoval() {
-    await this.pollCommand('docker network ls | grep wdio || true', (result) => (
-      new Promise((resolve, reject) => {
-        const { stdout: networkStatus } = result;
+  async startSeleniumHub() {
+    logger.info('Starting the docker selenium hub...');
 
-        // Reject if there is an active network returned.
-        if (networkStatus) {
-          reject();
-        } else {
-          resolve();
-        }
-      })));
-  }
+    const envVars = this.version ? `TERRA_SELENIUM_DOCKER_VERSION=${this.version} ` : '';
 
-  /**
-   * Ensures the docker services have been shut down.
-   */
-  async waitForServiceRemoval() {
-    await this.pollCommand('docker service ls | grep wdio || true', (result) => (
-      new Promise((resolve, reject) => {
-        const { stdout: serviceStatus } = result;
-
-        // Reject if there is an active service returned.
-        if (serviceStatus) {
-          reject();
-        } else {
-          resolve();
-        }
-      })));
+    await exec(`${envVars}docker-compose -f ${this.getDockerComposeFilePath()} up -d`);
+    await this.waitForSeleniumHubReady();
   }
 
   /**
-   * Ensures the docker network is ready.
+   * Waits for the docker selenium hub to become healthy.
    */
-  async waitForNetworkReady() {
-    logger.info('Waiting for the selenium grid to become ready.');
+  async waitForSeleniumHubReady() {
+    logger.info('Waiting for the docker selenium hub to become ready...');
 
-    await this.pollCommand(`curl -sSL http://${this.host}:${this.port}/wd/hub/status`, (result) => (
+    await this.pollCommand("docker inspect --format='{{json .State.Health.Status}}' selenium-hub", (result) => (
       new Promise((resolve, reject) => {
         const { stdout } = result;
-        const { value } = JSON.parse(stdout);
 
-        if (value.ready) {
+        if (stdout && stdout.trim() === '"healthy"') {
```

Comment:
Can just the single quote be used here or does it also need the double quotes?

Reply:
It needs the double quotes, the stdout pipes the values in as a quoted string from a JSON value. I decided to compare against that rather than manipulate the string (other than the trim, which is removing empty spaces and new line characters).

Reply:
Follow-up question: Since we're doing the health check on the grid once every 15 seconds, does that mean this may need to wait up to 15 seconds to get the up-to-date status? Otherwise, wouldn't the status be outdated? Or does executing `docker-compose -f` updates the health check status right away. Is that how it works?

Reply:
The healthcheck in the dockerfile will perform the check once every 15 seconds. When the compose file launches the status will be "starting" and it will immediately perform a healthcheck. If that healthcheck doesn't succeed it keeps trying for the number of `retries` every 15 seconds until it appears as healthy. If it grid was already up (say using the keep alive flag or the test run was hard killed) the grid will be healthy immediately because it is already running. The docker service is currently polling that health check every two seconds for up to two minutes. I don't think we have to worry about outdated statuses here, the tests won't launch unless a successful healthy check succeeds

Reply:
Thanks for the explanation!